com.unity.netcode.adapter.utp/Runtime/UnityTransport.cs

```diff
@@ -566,6 +571,73 @@ private void OnDestroy()
             DisposeDriver();
         }
 
+#if MULTIPLAYER_TOOLS
+        private void ExtractNetworkMetrics()
+        {
+            if (NetworkManager.Singleton.IsServer)
+            {
+                var ngoConnectionIds = NetworkManager.Singleton.ConnectedClients.Keys;
+                foreach (var ngoConnectionId in ngoConnectionIds)
+                {
+                    //Connection 0 when host is self and bypass the transport so would be invalid, skip it.
+                    if (ngoConnectionId > 0 || (ngoConnectionId == 0 && !NetworkManager.Singleton.IsHost))
+                    {
+                        ExtractNetworkMetricsForClient(NetworkManager.Singleton.ClientIdToTransportId(ngoConnectionId));
+                    }
+                }
+            }
+            else
+            {
+                ExtractNetworkMetricsForClient(NetworkManager.Singleton.ClientIdToTransportId(NetworkManager.Singleton.ServerClientId));
+            }
+        }
+
+        private void ExtractNetworkMetricsForClient(ulong transportClientId)
+        {
+            var networkConnection =  ParseClientId(transportClientId);
+
+            //Two pipelines, unreliable sequence and reliable fragmented
+            for (byte i = 0; i < k_PipelineCount; ++i)
```

Comment:
Why are you using `byte` as the type of the loop variable? Most likely the machine doesn't have registers that small in any case, and this will just result in greater risk of overflow without reducing resource usage 🤔

Reply:
I'll defer to @simon-lemay-unity for this. It does represent the small value well enough and I'm fine with it.

Reply:
I don't really have any strong opinion on the matter. Usually it's advised to use register-sized variables for loop counters since that's what will be fastest. But here it's a loop with 3 iterations, so unlikely to have any effect. If only for consistency with other loops in the code I'd go with an `int`.

Reply:
Still not sure I understand why we have a loop in the first place. Why isn't it just:
```
ProcessPipeline(m_UnreliableFragmentedPipeline);
ProcessPipeline(m_UnreliableSequencedFragmentedPipeline);
ProcessPipeline(m_ReliableSequencedFragmentedPipeline);
```

Reply:
Because I forgot your comment and this makes a lot more sense.